src/components/graph/network-modification-tree-model.ts

```diff
@@ -50,6 +98,9 @@ export default class NetworkModificationTreeModel {
             }
             case NodeInsertModes.NewBranch: {
                 // We need to insert the node after all children of the active(reference) node
+                if (!referenceNodeId) {
+                    break;
+                }
```

Comment:
i think it may introduce a bug if we change before and after. splice can work with even if reverenceNodeId is undefined
![image](https://github.com/user-attachments/assets/289bc552-6c81-4bae-ad6f-36a39c431283)